src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -501,13 +508,10 @@ def _process_checkin(item: CheckinItem, txn: Transaction | Span):
 
     monitor_config = params.pop("monitor_config", None)
 
-    params["duration"] = (
+    if params.get("duration") is not None:
         # Duration is specified in seconds from the client, it is
         # stored in the checkin model as milliseconds
-        int(params["duration"] * 1000)
-        if params.get("duration") is not None
-        else None
-    )
+        params["duration"] = int(params["duration"] * 1000)
```

Comment:
There's a small behavior change here right, since before it was setting `params["duration"]` to none even if it wasn't part of the dict. Now it will leave it not part of the dict.

not sure if this is actually important.

Reply:
there is yeah, but the types forbid `None` in this dict -- `params` are passed through a validator which normalizes it to `None`

afaict this still works since duration is only read from the validated params